API-65 Add property for end-user error messages (#79)
cultuurnet/apidocs: projects/errors/docs/format.md

# Response format & examples

> Error responses from publiq APIs follow the [RFC7807](https://datatracker.ietf.org/doc/html/rfc7807) standard, also known as "Problem Details for HTTP APIs".

Reviewer: Is it clear for client developers what "publiq APIs" are?

Maybe we need an extra section somewhere else (maybe on the doc home page) to clarify this. Or maybe we can clarify it here.

Author: This was just moved from existing documentation to a new page, and we use this in other places as well. I think it should become clear once we have "publiq platform", as well as from the URL (docs.publiq.be) and the logo in the top left (which mentions "publiq").

Author: But the home page will be expanded before we release publiq platform / these docs (to completely replace documentatie.uitdatabank.be) - so that can also further clarify it yes

*   `title` - A short, human-readable summary of the problem type. Note that this is usually targetted towards developers, not end-users.
*   `status` - The HTTP status code generated by the origin server for this occurrence of the problem.

Sometimes the following optional properties are also included:

*   `detail` - A human-readable summary about this specific instance of the problem. Note that this is usually targetted towards developers, not end-users.

For example:

```json
{
  "type": "https://api.publiq.be/probs/url/not-found",
  "title": "Not Found",
  "status": "404",
  "detail": "The event with id \"foobar\" was not found."
}
```

### Extensions

Aside from the standardized properties above, the RFC7807 spec also allows for custom properties. The following extra properties are sometimes used on publiq APIs to provide more information in specific situations.

#### endUserMessage

> Error responses may include an `endUserMessage` property with a message that explains the problem to an end-user, localized in one or more languages (depending on the target audience of the API). These are usually included in domain errors, for example when an UiTPAS passholder has an expired social tariff.

Reviewer: single end user message?

```suggestion
Error responses may include an `endUserMessage` property with a message that explains the problem to an end-user, localized in one or more languages (depending on the target audience of the API). It is usually included in domain errors, for example when an UiTPAS passholder has an expired social tariff.
```

> When an `endUserMessage` is included in the error response, it is strongly encouraged to always show these to end-users.

Reviewer: ```suggestion
When an `endUserMessage` is included in the error response, it is strongly encouraged to show it to the end-user.
```
